### ISSUE TITLE
Rechartering Web Machine Learning Working Group

### ISSUE THREAD

**Issue by dontcallmedom** [w3c/strategy#367]
New charter proposal, reviewers please take note.

# Charter Review

[2nd Web Machine Learning Working Group Charter](https://w3c.github.io/machine-learning-charter/charter.html)

What kind of charter is this? Check the relevant box / remove irrelevant branches. 
* Existing 
 - [x] Existing WG recharter  - [diff](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2F2021%2F04%2Fweb-machine-learning-charter.html&doc2=https%3A%2F%2Fw3c.github.io%2Fmachine-learning-charter%2Fcharter.html)
 - [ ] Existing IG recharter
 - If this is a charter extension or revision, link a [diff from previous charter](https://services.w3.org/htmldiff), and any issue discussion:

Communities suggested for outreach: 
* machine-learning based application providers, including live media processing, speech recognition engines

Known or potential areas of concern: 
* at the moment, implementation work is only happening in the Chromium project, with no clear signals from other browser engines on their intent to implement this; this is mitigated by:
  * the fact that the back-ends the API needs to plug into are very different, and the majority of the work in implementing the API lies into this adaptation, which has already provided significant diversity in implementation experience feedback
  * the performance gains provided by the API (compared to existing capabilities of the platform through WebAssembly and WebGPU) are impressive in very compelling scenarios such as [live video processing in real-time communication contexts](https://github.com/webmachinelearning/webnn/issues/226)
  * the usage of machine learning models is overall exploding and providing a strong basis in the Web Platform is more important than ever
* the interaction with GPUs is tied to the work happening in the GPU on the Web Working Group, which so far has had limited bandwidth to review WebNN/WebGPU integration; as the WebGPU API stabilizes, the Machine Learning Working Group is hoping to get more of their attention
* usage of machine learning models can create ethical challenges; the Working Group is addressing it through its work on [ethical principles for machine learning on the Web](https://webmachinelearning.github.io/ethical-webmachinelearning/)

Where would charter proponents like to see issues raised? (this strategy funnel issue, a different github repo, email, ...):
https://github.com/w3c/machine-learning-charter/

Anything else we should think about as we review?

**Comment by michael-n-cooper** [comment]
I'm not sure about the section 3.2 Tentative Specifications, which is promising a deliverable which is under incubation but has not yet matured to a point it is ready for AC review to include in a WG charter. This could lead to the group working on a document which the AC has approved in principle but has not actually seen, and given the active incubation we don't know how the initial reference draft will look. I'm inclined to think that either the deliverable should be promoted as a full deliverable, or mentioned as an "other deliverable" that may get worked on and chartered in a future round.

I recognize that the charter was approved in this form previously. I am raising this from a new role as part of the TiLT team, and is just a question I think merits consideration.

Aside from this question, I think the charter looks good.

**Comment by ylafon** [comment]
In 5.1 W3C Groups, is there a specific reason to link to the WASM CG and not WG?
I second @svgeesus comment about updating to the most recent charter template for Design Principles.

Considering this is a recharter, it looks good.

**Comment by r12a** [comment]
Personal comment: The areas in scope appeared a little lop-sided to me, since Speech Recognition is called out but not Speech Synthesis.  Specifically, just as you need to infer text from wave input in speech recognition, you need to infer meaning and pronunciation guides (text understanding) as a prep for speech synthesis, and i wonder why  machine learning is not being applied to that and included in the scope of the charter (ie. for the linguistic analysis of the text, rather than what drives the audio hardware, and could provide _input_ to other TTS specs at the W3C - such as SSML, CSS, etc). It seems that doesn't fall within the scope of the work, but I think the spec should probably indicate briefly why it is not addressing the use of machine learning for that function, whereas it does make inferences for speech recognition.

**Comment by himorin** [comment]
I18n hopes that the machine will be capable of detecting language and direction and applying metadata to the API for those things to be available for the output text. 
We're not sure that this needs a charter change, but would like to remind the group at this early stage that it is one of the key things we'll be looking for when we come to review the specs.

**Comment by svgeesus** [comment]
> I'm not sure about the section 3.2 Tentative Specifications, which is promising a deliverable which is under incubation but has not yet matured to a point it is ready for AC review to include in a WG charter.

I see your point but I read that as _not_ promising a deliverable. Instead it is placed in scope so that, if it matures sufficiently, it can be worked on. That seems sensible to me, and there is a draft so AC can look at it and object to it's inclusion if they want.

**Comment by ruoxiran** [comment]
APA WG reviewed this charter, looks same as previous, and have no comments.

**Comment by dontcallmedom** [comment]
@ylafon re
> In 5.1 W3C Groups, is there a specific reason to link to the WASM CG and not WG?

My understanding is that the gist of the work is happening in the CG rather than the WG, so coordinating with the CG feels like a better reflection of what needs to happen.

**Comment by dontcallmedom** [comment]
I've submitted a PR to charter to add the explicit reference to the TAG design principles https://github.com/w3c/machine-learning-charter/pull/30 (which the WG already strives to adhere to in any case).

My reading of the other comments is that they don't need a charter change, so at this point, I'll investigate getting that charter under AC review

**Comment by r12a** [comment]
I just realised that my comment of 2 weeks ago contained a misprint.

>  It seems that doesn't fall within the scope of the work, but I think the spec should probably indicate briefly why it is not addressing the use of machine learning for that function, whereas it does make inferences for speech recognition.

I meant to say the charter, not the spec (since this is to do with the scope of the work).

**Comment by dontcallmedom** [comment]
@r12a thanks for the clarification - my reading of the examples in "motivation and background" is that they illustrate what impact the work is expected to have, they're not meant as a limitation to its impact. My superficial knowledge of ML & Speech Synthesis tells me that WebNN would also help with models used to optimize speech synthesis, although there are likely be additional challenges in deploying speech synthesis beyond optimization of computation.

If you think speech synthesis should be explicitly mentioned in "motivation and background", could you bring this to https://github.com/w3c/machine-learning-charter/issues where people with more expertise than I do would be able to comment on whether and how to include it there.

**Comment by plehegar** [comment]
> @ylafon re
> 
> > In 5.1 W3C Groups, is there a specific reason to link to the WASM CG and not WG?
> 
> My understanding is that the gist of the work is happening in the CG rather than the WG, so coordinating with the CG feels like a better reflection of what needs to happen.

This is correct indeed.

**Comment by plehegar** [comment]
announced: https://lists.w3.org/Archives/Member/w3c-ac-members/2023AprJun/0003.html
